File: driver/test/CPP/node.cpp
/*******************************************************************************

   Copyright (C) 2011-Present SequoiaDB Ltd.

   Licensed under the Apache License, Version 2.0 (the "License");
   you may not use this file except in compliance with the License.
   You may obtain a copy of the License at

      http://www.apache.org/licenses/LICENSE-2.0

   Unless required by applicable law or agreed to in writing, software
   distributed under the License is distributed on an "AS IS" BASIS,
   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
   See the License for the specific language governing permissions and
   limitations under the License.

   Source File Name = node.cpp

   Descriptive Name = N/A

   Dependencies: N/A

   Restrictions: N/A

   Change Activity:
   defect Date        Who         Description
   ====== =========== =========== =============================================
          23/07/2025  fangjiabin  Initial Draft

   Last Changed =

*******************************************************************************/
#include <stdio.h>
#include <gtest/gtest.h>
#include "client.hpp"
#include "testcommon.hpp"
#include "deleteFile.hpp"
#include <string>
#include <iostream>

using namespace std ;
using namespace sdbclient ;

/*************************************
 *           test begine             *
 *************************************/

TEST(node, not_connect)
{
   sdbNode node ;
   BSONObj obj ;
   INT32 rc = SDB_OK ;

   rc = node.getStatus() ;
   ASSERT_EQ( SDB_NOT_CONNECTED, rc ) ;
}

TEST(node,getStatus)
{
   sdb connect ;
   sdbShard shard ;
   sdbNode node ;
   sdbCursor cursor ;
   // initialize local variables
   const CHAR *pHostName                    = HOST ;
   const CHAR *pPort                        = SERVER ;
   const CHAR *pUsr                         = USER ;
   const CHAR *pPasswd                      = PASSWD ;
   INT32 rc                                 = SDB_OK ;
   // connect to database
   rc = connect.connect( pHostName, pPort, pUsr, pPasswd ) ;
   ASSERT_EQ( SDB_OK, rc ) ;
   rc = connect.getShard("group1", shard) ;
   ASSERT_EQ( SDB_OK, rc ) ;
   rc = shard.getNode("ubuntu-dev1", "51000", node);
   ASSERT_EQ( SDB_OK, rc ) ;
   sdbNodeStatus status = SDB_NODE_UNKNOWN;
   status = node.getStatus();
   cout << "status = " << status << endl ;
   // disconnect the connection
   connect.disconnect() ;
}

// TODO:
//all

File: misc/ci/ant-sdbtask/src/com/sequoiadb/ant/datatype/Node.java
/*******************************************************************************

   Copyright (C) 2011-Present SequoiaDB Ltd.

   Licensed under the Apache License, Version 2.0 (the "License");
   you may not use this file except in compliance with the License.
   You may obtain a copy of the License at

      http://www.apache.org/licenses/LICENSE-2.0

   Unless required by applicable law or agreed to in writing, software
   distributed under the License is distributed on an "AS IS" BASIS,
   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
   See the License for the specific language governing permissions and
   limitations under the License.

   
*******************************************************************************/
package com.sequoiadb.ant.datatype;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

//import org.apache.tools.ant.types.*;

public class Node {
	private String host;
	private String baseport;
	//private String catalogaddr;
	private String groupname;
	private String dbpath;
	
	List<ConfItem> confItems = new ArrayList<ConfItem>();
	
	public void setDbpath(String value)
	{
		dbpath = value;
	}
	
	public String getDbpath()
	{
		return dbpath;
	}
	
	public void setHost(String value)
	{
		host = value;
	}
	
	public String getHost()
	{
		return host;
	}
	
	public void setBaseport(String value)
	{
		baseport = value;
	}
	
	public int getBasePort()
	{
		return Integer.parseInt(this.baseport);
	}
	
	//public void setCatalogaddr(String value)
	//{
	//	catalogaddr = value;
	//}
	
	//public String getCatalogaddr()
	//{
	//	return catalogaddr;
	//}
	
	public void setGroupname(String value)
	{
		groupname = value;
	}
	
	public String getGroupName()
	{
		return groupname;
	}
	
	public ConfItem createConfitem()
	{
		ConfItem item = new ConfItem();
		confItems.add(item);
		return item;
	}
	
	public Map<String, String> getConfigMap()
	{
		Map<String, String> configItemsMap = new HashMap<String,String>();
		
		for (ConfItem item : confItems)
		{
			configItemsMap.put(item.getName(), item.getValue());
		}
		return configItemsMap;
	}
	
	public String toString()
	{
		String msg = "{dataNode{";
		msg += "baseport:" + baseport + ",";
		//msg += "catalogaddr:" + catalogaddr + ",";
		msg += "groupname:" + groupname + ",";
		msg += "configtimes{";
		
		for(ConfItem item: confItems)
		{
			msg += item.toString() + ",";
		}
		
		msg += "}";
		
		return msg;
	}
	
}
